src/components/__snapshots__/FilterSideBar.test.js.snap

```diff
@@ -24,10 +24,10 @@ exports[`<FilterSideBar/> Matches the snapshot for category: TOP_FAVORITES 1`] =
   >
     Usage Analytics
   </h1>
-  <Connect(Category) />
-  <Connect(EventType) />
-  <Connect(PageSize) />
-  <Connect(SortOrder) />
+  <ConnectFunction />
+  <ConnectFunction />
+  <ConnectFunction />
+  <ConnectFunction />
```

Comment:
Prior to this dependency upgrade, it was possible to see exactly which connected components would be included in the snapshot. Now they all look the same, which makes the snapshot a bit meaningless on the human eye.

In the current setup, because each scenario renders a different number of components, the tests still "work" to some extend. But in theory, the right number of components could be rendered, but in the wrong sequence or of the wrong type, and the tests would not pick up on this.

So, we should look into fixing this. I see two possible solutions:
1. Fix the test so the connected component names show up again
1. If 1 is not possible, we can also adopt a different strategy for testing the same thing.

Reply:
That's true, I'll look into it.

Reply:
I've been looking into a suitable replacement for this. It's possible for example to assert with enzyme that an element at a certain location in the tree is the exact expected element.

However, it's a bit awkward (I can't get it to work exactly the way I'd want to with enzyme). And since you're going down the component tree quite a bit, it feels like a bit of a pointless test. Rather than a shallow snapshot, you're confirming that at x position in the tree there's an element you expect. I think it's not that necessary to test that. So I'm fine with leaving the tests as is.

Let me know if you think otherwise though.

Reply:
Actually, one thing that might resolve this cleanly is a custom snapshot serializer. I'll look into that.

Reply:
Allright, implemented the `react-redux` mock like you suggested. The only slightly fiddly thing is the mocking for the `<App />` component. We could mock it for all tests, but I'd prefer to use the real implementation where possible. But it works, and the mocking/unmocking is legible I think.

Let me know what you think.

Reply:
I agree it is better to keep the real implementation where possible. It would have been nice if could have perhaps extracted this to some test helper. But unfortunately, the mock for the App component is slightly different from the other two. As such, I am happy with the current implementation